store/arc.go

```diff
@@ -301,73 +300,47 @@ func (s *store) btGetPropertyValues(ctx context.Context,
 		rowRangeList = append(rowRangeList, bigtable.PrefixRange((rowPrefix)))
 	}
 
-	btRawValuesMapChan := make(chan map[string][]string, len(rowRangeList)/1000+1)
-	errs, errCtx := errgroup.WithContext(ctx)
-	for i := 0; i <= len(rowRangeList)/1000; i++ {
-		left := i * 1000
-		right := (i + 1) * 1000
-		if right > len(rowRangeList) {
-			right = len(rowRangeList)
-		}
-
-		errs.Go(func() error {
-			btRawValuesMap := map[string][]string{} // Key: dcid; value: a list of raw row values.
-
-			// RowSet of ReadRows have size limit of 500k.
-			if err := btTable.ReadRows(errCtx, rowRangeList[left:right],
-				func(btRow bigtable.Row) bool {
-					// Extract DCID from row key.
-					rowKey := btRow.Key()
-					parts := strings.Split(rowKey, "-")
-					dcid := strings.TrimPrefix(parts[0], keyPrefix(arcOut))
-
-					btResult := btRow[util.BtFamily][0]
-					if _, ok := btRawValuesMap[dcid]; !ok {
-						btRawValuesMap[dcid] = []string{}
-					}
-					btRawValuesMap[dcid] = append(btRawValuesMap[dcid], string(btResult.Value))
+	btRawValuesMap := map[string][]string{} // Key: dcid; value: a list of raw row values.
+	if err := util.BigTableReadRowsParallel(ctx, s.btClient, rowRangeList,
+		func(btRow bigtable.Row) error {
+			// Extract DCID from row key.
+			rowKey := btRow.Key()
+			parts := strings.Split(rowKey, "-")
+			dcid := strings.TrimPrefix(parts[0], keyPrefix(arcOut))
 
-					return true
-				}); err != nil {
-				return err
+			btResult := btRow[util.BtFamily][0]
+			if _, ok := btRawValuesMap[dcid]; !ok {
+				btRawValuesMap[dcid] = []string{}
 			}
-
-			btRawValuesMapChan <- btRawValuesMap
+			btRawValuesMap[dcid] = append(btRawValuesMap[dcid], string(btResult.Value))
 
 			return nil
-		})
-	}
-
-	err := errs.Wait()
-	if err != nil {
+		}); err != nil {
 		return nil, err
 	}
-	close(btRawValuesMapChan)
 
 	nodeRes := make(map[string]map[string][]Node, 0)
-	for btRawValuesMap := range btRawValuesMapChan {
-		for dcid, btRawValues := range btRawValuesMap {
-			nodeList := make([]Node, 0)
-			for _, btRawValue := range btRawValues {
-				btJSONRaw, err := util.UnzipAndDecode(string(btRawValue))
-				if err != nil {
-					return nil, err
-				}
+	for dcid, btRawValues := range btRawValuesMap {
+		nodeList := make([]Node, 0)
+		for _, btRawValue := range btRawValues {
+			btJSONRaw, err := util.UnzipAndDecode(string(btRawValue))
```

Comment:
Maybe should start test if we just save the original data and save this step.

Reply:
Yes, for GetPropertyValues, we do plan to remove compression and b64 encoding. Will change this when BT data is updated.